src/modules/sensors/rc_update.cpp

```diff
@@ -349,26 +349,41 @@ RCUpdate::rc_poll(const ParameterHandles &parameter_handles)
 				/* number of valid slots */
 				const int num_slots = manual_control_setpoint_s::MODE_SLOT_NUM;
 
-				/* the half width of the range of a slot is the total range
-				 * divided by the number of slots, again divided by two
-				 */
-				const float slot_width_half = 2.0f / num_slots / 2.0f;
-
-				/* min is -1, max is +1, range is 2. We offset below min and max */
-				const float slot_min = -1.0f - 0.05f;
-				const float slot_max = 1.0f + 0.05f;
-
-				/* the slot gets mapped by first normalizing into a 0..1 interval using min
-				 * and max. Then the right slot is obtained by multiplying with the number of
-				 * slots. And finally we add half a slot width to ensure that integer rounding
-				 * will take us to the correct final index.
-				 */
-				manual.mode_slot = (((((_rc.channels[_parameters.rc_map_flightmode - 1] - slot_min) * num_slots) + slot_width_half) /
-						     (slot_max - slot_min)) + (1.0f / num_slots)) + 1;
-
-				if (manual.mode_slot > num_slots) {
-					manual.mode_slot = num_slots;
-				}
+				/* determine flight mode slot index from rc input */
+  				float rc_channel_val = _rc.channels[_parameters.rc_map_flightmode-1];
+
+  				/* each 0.1 represent 50us of PWM.  */
+  				if (rc_channel_val >= -1.05f && rc_channel_val < -0.68f) {  //PWM 1000 - 1160
```

Comment:
I'm not sure about these magic numbers. Maybe we can find a line inbetween the heavily condensed convoluted equation unrolling it with magic numbers?

Reply:
Yeah, exactly, I had the same idea as I was putting together the PR.  Just wanted some feedback if this is viable before I put more effort in.

I think if we just map _rc.channels into pwm space then the comparisons would be on us of PWM.  So the first if statement would be 

`if (rc_channel_val >= 1000f && rc_channel_val < 1160)`

Reply:
@dagar let me know you opinion if this works better for determining the pwm bands.